File: CSharp Part 2/StringsAndTextProcessingHomework/Problem 02. Reverse string/ReverseString.cs
// Problem 2. Reverse string
// Write a program that reads a string, reverses it and prints the result at the console.

using System;
using System.Text;

class ReverseString
{
    static void Main()
    {
        Console.WriteLine("This program reads a text, reverses it and prints the result at the console.");
        Console.Write("Please enter text: ");
        string input = Console.ReadLine();
        StringBuilder reverseText = new StringBuilder(input.Length);
        for (int i = input.Length - 1; i >= 0; i--)
        {
            reverseText.Append(input[i]);
        }
        Console.WriteLine("The reversed text is: {0}", reverseText);
    }
}

File: JavaScript Fundamentals/StringsHomework/Problem 1. Reverse string/ReverseString.js
if (!String.prototype.reverse) {
    String.prototype.reverse = function () {
        var i,
            len = this.length,
            result = '';
        for (i = len - 1; i >= 0; i -= 1) {
            result += this[i];
        }
        return result;
    }
}

var someString = 'Some string',
    reversed;

reversed = someString.reverse();

console.log(reversed);

